docs/articles/basic_building_blocks.md

```diff
@@ -0,0 +1,121 @@
+
+Let’s take a look at how these building blocks do this in a little more detail.
+
+## Turning classes into Schemas
+
+Once you’ve parsed data into your notebook via JSON or a pandas dataframe, it’s time to create a Schema describing your data.
+
+To do this, you **use the Schema decorator to annotate your class as a schema** representing your structured data. Schemas translate to searchable entities in the embedding space. To get started, type @schema, and then define the field types to match the different types of data you’ve imported.
+
+```python
+@schema
+class ParagraphSchema:
+    body: String
+    id: IdField
+```
+
+With your Schemas created, you are ready to move on to embedding and querying, which is where Superlinked’s building blocks approach really empower you. The Superlinked framework is based on the intuition that people doing semantic search using embedding vectors can better satisfy the requirements of their use case/s if they can customize how their system handles data and queries. **Spaces** is a declarative class developed with this in mind. The Space module encapsulates the vector creation logic that will be used at ingestion time, and again at query time.
+
+## Declaring how to embed your data using Spaces
+
+Spaces lets you tailor how you embed different attributes of your data. Spaces can be categorized along 2 key dimensions:
+1. what input types the Space permits - e.g., text, timestamp, numeric, categorical
+2. whether the Space represents similarity (e.g, TextSimilaritySpace) or scale (e.g., numeric space)
+Which Space/s fit your use case depends on both these dimensions - your input type and what you need to represent about your data.
+
+You use different Spaces for different data types. For example, TextSimilaritySpace and CategoricalSimilaritySpace can take String as an input, RecencySpace can take Timestamp, NumberSpace can take Int/Float, and so on. Each Space captures a different, relevant piece of information (e.g., title, review count, etc.) about an entity. This lets you weight each Space according to which attributes are relevant to your use case - before you concatenate all your Spaces into a single multimodal vector among others in the queryable vector space.
+
+By prioritizing the creation of smarter vectors up front - and only then creating the index - we can achieve better quality retrieval, without costly and time-consuming reranking and general post-processing work.
+
+```python
+relevance_space = TextSimilaritySpace(text=paragraph.body, model="all-MiniLM-L6-v2")
+```
+
+## Indexing
+
+Superlinked’s Index module components enable you to group Spaces into indices that make your queries more efficient.
+
+```python
+paragraph_index = Index(relevance_space)
+```
+
+## Executing your query to your chosen endpoints
+
+Before running your code, you need to structure your query using the following arguments:
+- Query: defines the index you want it to search
```

Comment:
let's link feature notebooks here?

also it defines the index and you can also add Params here.
param feature nb: https://github.com/superlinked/superlinked/blob/main/notebook/feature/dynamic_parameters.ipynb